zio-http/src/test/scala/zhttp/http/URLSpec.scala

```diff
@@ -99,6 +99,38 @@ object URLSpec extends DefaultRunnableSpec {
     },
   )
 
+  val builderSpec = suite("builder")(
+    test("creates a URL with all attributes set") {
+      val builderUrl = URL.empty
+        .setHost("www.yourdomain.com")
+        .setPath("/list")
+        .setPort(8080)
+        .setScheme(Scheme.HTTPS)
+        .setQueryParams("?type=builder&query=provided")
+
+      val constructedUrl =
+        URL(
+          Path("/list"),
+          URL.Location.Absolute(Scheme.HTTPS, "www.yourdomain.com", 8080),
+          Map("type" -> List("builder"), "query" -> List("provided")),
+        )
+
+      assert(builderUrl)(equalTo(constructedUrl))
```

Comment:
Instead of comparing the ADTs for equality, we should encode it as string and then compare with with the expected string.
```suggestion
      assert(builderUrl.encode)(equalTo("https://www.yourdomain.com:8080/list?typebuilder&query=provided"))
```